gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/AndroidPluginIntegration.kt

```diff
@@ -68,28 +70,48 @@ object AndroidPluginIntegration {
      */
     @Suppress("DEPRECATION")
     private fun tryUpdateKspWithAndroidSourceSets(
+        project: Project,
         kotlinCompilation: KotlinJvmAndroidCompilation,
         kspTaskProvider: TaskProvider<*>
     ) {
+        val kaptProvider: TaskProvider<Task>? =
+            project.locateTask(kotlinCompilation.compileTaskProvider.kaptTaskName)
+
         kotlinCompilation.androidVariant.getSourceFolders(SourceKind.JAVA).forEach { source ->
             kspTaskProvider.configure { task ->
-                when (task) {
-                    is KspTaskJvm -> {
-                        task.setSource(source)
-                        task.dependsOn(source)
-                    }
+                // this is workaround for KAPT generator that prevents circular dependency
+                val isKaptOutput =
+                    kaptProvider?.let { it.get().outputs.files.toList().any { file -> source.dir == file } } ?: false
```

Comment:
1. `get()` at configuration time forces resolution of the KAPT task. Is it possible to filter `source` in a callable, and pass the callable to `task.setSource` and `task.kspConfig.javaSourceRoots.from` below?
2. Do those files always match? E.g., KAPT's `outputs.files` be regular files instead of directories?
3. Some properties of kapt task are API. For example, `destinationDir` is guaranteed to be a directory. If 2. is an issue, would it be easier to use?

Reply:
Thanks for feedback, I wrapped getter to Callable and use KaptTask.destinationDir to have generated folder we look  for.